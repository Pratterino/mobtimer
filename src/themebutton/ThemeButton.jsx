import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { setTheme } from './../settings/settingsActions';
import './ThemeButton.scss';

function ThemeButton(props) {
    return (
        <div className="theme-button">
            <button
                key={props.theme}
                className="theme-button__button"
                onMouseLeave={() => props.setTempTheme(null)}
                onMouseOver={() => props.setTempTheme(props.theme)}
                onClick={() => props.setTheme(props.theme)}>
                {props.theme}
            </button>
        </div>
    );
}

const mapStateToProps = state => ({
    settings: state.settings,
});

const mapDispatchToProps = dispatch =>
    bindActionCreators(
        {
            setTheme,
        },
        dispatch,
    );

ThemeButton.propTypes = {
    setTheme: PropTypes.func.isRequired,
    setTempTheme: PropTypes.func.isRequired,
};

export const unwrapped = ThemeButton;
export default connect(mapStateToProps, mapDispatchToProps)(ThemeButton);
