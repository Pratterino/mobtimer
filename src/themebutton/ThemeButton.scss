.theme-button {
    display: inline-block;
    margin: 2px;

    .theme-button__button {
        &:hover {
            cursor: help;
        }
    }
}
