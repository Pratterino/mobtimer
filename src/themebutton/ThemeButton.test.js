import React from 'react';
import { unwrapped as UnwrappedThemeButton } from './ThemeButton';
import { shallow } from 'enzyme';

describe('ThemeButton', () => {
    let wrapper;

    const renderComponent = props =>
        (wrapper = shallow(
            <UnwrappedThemeButton
                settings={{
                    theme: 'ducca-evening',
                }}
                {...props}
            />,
        ));

    beforeEach(() => {
        renderComponent();
    });

    describe('render', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
