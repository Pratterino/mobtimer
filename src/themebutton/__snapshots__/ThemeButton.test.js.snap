// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeButton render should match snapshot 1`] = `
<div
  className="theme-button"
>
  <button
    className="theme-button__button"
    onClick={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  />
</div>
`;
